cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,27 +1,36 @@
-describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("[name=SubmitLogin]").click();
-    cy.get(".cart_navigation span").click();
+import {
+  MenuContentPage,
+  ProductslistPage,
+  ShopingCartPage,
+  LoginPage,
+  AdressStepPage,
+  ShippingStepPage,
+  PaymentStepPage,
+} from "../page/index";
 
-    cy.get("[name=cgv]").click();
-    cy.get(".cart_navigation span").click();
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductslistPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const adressStepPage = new AdressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
 
-    cy.get("[title='Pay by bank wire']").click();
-    cy.get(".cart_navigation span").click();
-
-
-    cy.get("#center_column > div > p > strong").should(
+describe("Buy a t-shirt", () => {
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+    productsListPage.goToProductList();
+    productsListPage.proceedToCheckout();
+    shopingCartPage.proceedToCheckout();
+    loginPage.signIn("aperdomobo@gmail.com", "WorkshopProtractor");
+    loginPage.goToSubmit();
```

Comment:
this method is not needed, the click on the submit button also can be within the signIn method.